tree/tree/src/TTreeCloner.cxx

```diff
@@ -691,13 +729,28 @@ void TTreeCloner::WriteBaskets()
       TBranch *from = (TBranch*)fFromBranches.UncheckedAt( fBasketBranchNum[ fBasketIndex[j] ] );
       TBranch *to   = (TBranch*)fToBranches.UncheckedAt( fBasketBranchNum[ fBasketIndex[j] ] );
 
-      TFile *tofile = to->GetFile(0);
+      TFile *tofile = fToFile;
       TFile *fromfile = from->GetFile(0);
 
       Int_t index = fBasketNum[ fBasketIndex[j] ];
 
       Long64_t pos = from->GetBasketSeek(index);
-      if (pos!=0) {
+      if (IsInPlace()) {
+         if (pos != 0) {
+            if (fFileCache && j >= notCached) {
+               notCached = FillCache(notCached);
+            }
+            if (from->GetBasketBytes()[index] == 0) {
+               from->GetBasketBytes()[index] = basket->ReadBasketBytes(pos, fromfile);
+            }
+            Int_t len = from->GetBasketBytes()[index];
+
+            basket->LoadBasketBuffers(pos,len,fromfile,fFromTree);
+            basket->IncrementPidOffset(fPidOffset);
+            basket->CopyTo(tofile);
+            to->fBasketSeek[index] = basket->GetSeekKey();
+         }
+      } else if (pos!=0) {
```

Comment:
You can avoid checking for `pos != 0` twice if you use `if (pos == 0) { ... } else if(IsInPlace()) { ... } else { ... }`

Reply:
I think that is not strictly equivalent.  The current scheme is
```
if (IsInPlace())
   if (pos != 0)
       something
   else
       nothing
else if (pos != 0)
   something else
else
   something else
```